lib/updates-panel.js

```diff
@@ -171,13 +171,18 @@ export default class UpdatesPanel {
           if (successfulUpdatesCount > 1) {
             pluralizedPackages += 's'
           }
-          const message = `Restart Atom to complete the update of ${successfulUpdatesCount} ${pluralizedPackages}.`
+          const message = `Restart Atom to complete the update of ${successfulUpdatesCount} ${pluralizedPackages}:`
+          let detail = ''
+          this.packageCards.forEach((card, index, array) => {
+            detail += `${card.pack.name}@${card.pack.version} -> ${card.pack.latestVersion}`
+            index === array.length - 1 ? null : detail += `\n`
```

Comment:
I think this is potentially clearer as `if (index !== array.length - 1) detail += '\n'`.  Alternatively, you could add the `\n` onto `detail` every time and then just `detail.trim()` after the forEach, which removes the need for us to pass in `index` and `array`.

Reply:
Good suggestion. Having an expression with side effects inside of a ternary is pretty unusual.

Reply:
Glad you seem to like it 😄
Could you confirm that what you are refering to as 'side effects' means changing a variable, so ternary expressions are meant for simple things and I should have used a good old if statement? (I'm beginner 👋 )

Reply:
Well in this case the majority of the time you're hitting the "null" statement, while if you used an if-statement you wouldn't have to explicitly specify "do nothing".  In that way if you ever find yourself not using part of the ternary statement I think that's a signal to use an if instead.
(Though in this scenario we were able to also remove the if through `detail.trim()`)